src/components/NewCommentForm/NewCommentForm.jsx

```diff
@@ -1,39 +1,90 @@
-import React from 'react';
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { addPostComment } from '../../api/comments';
 import './NewCommentForm.scss';
 
-export const NewCommentForm = () => (
-  <form className="NewCommentForm">
-    <div className="form-field">
-      <input
-        type="text"
-        name="name"
-        placeholder="Your name"
-        className="NewCommentForm__input"
-      />
-    </div>
+export const NewCommentForm = ({ id, getPostComments, setComments }) => {
+  const [newComment, setNewComment] = useState({
+    name: '',
+    email: '',
+    body: '',
+  });
 
-    <div className="form-field">
-      <input
-        type="text"
-        name="email"
-        placeholder="Your email"
-        className="NewCommentForm__input"
-      />
-    </div>
+  return (
+    <form
+      className="NewCommentForm"
+      onSubmit={(event) => {
+        event.preventDefault();
+        newComment.postId = id;
+        if (newComment.name === ''
+          || newComment.email === ''
+          || newComment.body === '') {
+          return;
+        }
 
-    <div className="form-field">
-      <textarea
-        name="body"
-        placeholder="Type comment here"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <button
-      type="submit"
-      className="NewCommentForm__submit-button button"
+        addPostComment(newComment)
+          .then(() => getPostComments(id)
+            .then(setComments));
+        setNewComment({
+          name: '',
+          email: '',
+          body: '',
+        });
+      }}
     >
-      Add a comment
-    </button>
-  </form>
-);
+      <div className="form-field">
+        <input
+          type="text"
+          name="name"
+          value={newComment.name}
+          placeholder="Your name"
+          className="NewCommentForm__input"
+          onChange={event => setNewComment({
+            ...newComment,
+            name: event.target.value,
+          })}
```

Comment:
```suggestion
          onChange={({ target }) => setCommentPart(target.name, target.value)}
```
create `setCommentPart` method, and pass it in every input. It should work like this:
```javascript
const setCommentPart = (partName, partValue) => {
  setNewComment(prevComment => ({
    ...prevComment,
    [partName]: partValue,
  }))
} 
```

Reply:
+